binary_gentoo/internal/atoms.py

```diff
@@ -17,5 +19,12 @@ def extract_category_package_from(atomlike):
         if match is not None:
             break
     else:
-        raise ValueError(f'Not valid "{ATOM_LIKE_DISPLAY}" syntax: {atomlike!r}')
+        raise ValueError(f'Not valid "{ATOM_LIKE_DISPLAY}" syntax for package: {atomlike!r}')
     return match.group('category'), match.group('package')
+
+
+def extract_set_from(set_candidate):
+    match = re.compile(_set_pattern).match(set_candidate)
+    if match is None:
+        raise ValueError(f'Not valid "{SET_DISPLAY}" syntax for set: {set_candidate!r}')
```

Comment:
For both "for package" and "for set", can we either:
- Make them "for a package" and "for a set" or
- drop those parts?

I suggesting that because if something is not a package, saying "for package: not-a-package" seems contradictory. By adding the "a", we are now speaking of a class rather than a single thing, at least that's the idea. Does that make sense?